core/domain/app_feedback_report_jobs_one_off.py

```diff
@@ -0,0 +1,53 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""One-off jobs for app feedback report models."""
+
+from __future__ import absolute_import # pylint: disable=import-only-modules
+from __future__ import unicode_literals # pylint: disable=import-only-modules
+
+import datetime
+
+from core import jobs
+from core.domain import app_feedback_report_services
+from core.platform import models
+import feconf
+
+(app_feedback_report_models,) = models.Registry.import_models([
+    models.NAMES.app_feedback_report])
+
+
+class ScrubReportsOneOffJob(jobs.BaseMapReduceOneOffJobManager):
```

Comment:
This name is a bit too generic. Maybe instead: ``ScrubAppFeedbackReportsOneOffJob`` might help clarify?

Reply:
+1, agree with @BenHenning. 

We might have a bit of a problem, since @brianrodri and @vojtechjelinek are actually moving all the jobs to Beam and we'll be deleting all MR jobs imminently. I wonder if this needs to be written in the updated Beam framework (assuming it's sufficiently mature at this point), or if it's OK to go ahead with this? /cc @brianrodri @vojtechjelinek to advise on this.

Reply:
It depends on what release are we aiming this to, if April it can still be written in MapReduce (but will need to be rewritten in Beam), if we aim this later it doesn't make sense to do this in MapReduce and we should wait fro Beam.

Reply:
Ah ok. @jcqli @BenHenning when will this job first need to be run?

Reply:
Not for awhile since storage won't be populated for some time and this job acts on models another 3 months after they're stored. Seems like we'll want to use Beam, then?

Reply:
Yup, I think so. Sg, please check with @vojtechjelinek if you need any guidance on the Beam stuff!

Reply:
Are there any existing examples for this?

Reply:
Right now we do not have an easy example and the architecture is not yet finalized. Probably wait around a month, then we should have some examples that you can follow.

Reply:
We won't be deleting any models, scrubbing is to scrub some of the data stored in the models (e.g. data that the user enters themselves from a text box).

 Discussed with Ben earlier and it's unlikely we'll hit 1000 reports but I don't think this would work the same way as `delete_models_marked_as_deleted` because `delete_models_marked_as_deleted` uses delete_multi() and we don't have an equivalent.

I think it's fine to do something similar but it would be queried / scrubbed a bit differently than `delete_models_marked_as_deleted`. In any case we can avoid the MapReduce but will need to double check that with Ben. Is this what you're suggesting to use cron_services for?

Reply:
Yes, it should be possible to rewrite this in a way that we perform a query and then srub the stuff in a for-loop and save the models again by using `put_multi`. The code can be placed in some service and that service can be called from the cron controller directly. The CRON jobs have 30 seconds until they timeout so there is a plenty of time.